aiohttp/web.py

```diff
@@ -277,6 +280,130 @@
 
 HostSequence = TypingIterable[str]
 
+if sys.version_info >= (3, 11):
+
+    class _State(enum.Enum):
+        CREATED = "created"
+        INITIALIZED = "initialized"
+        CLOSED = "closed"
+
+    class WebRunner(asyncio.Runner):  # type: ignore
+        """A context manager that controls event loop life cycle"""
+
+        def __init__(
+            self,
+            *,
+            debug: Optional[bool] = None,
+            loop_factory: Optional[Callable[[], asyncio.AbstractEventLoop]] = None,
+        ):
+            super().__init__(debug=debug, loop_factory=loop_factory)
+
+        def close(self) -> None:
+            """Shutdown and close event loop."""
+            if self._state is not _State.INITIALIZED:
+                return
+            loop = self._loop
+            try:
+                _cancel_tasks(tasks.all_tasks(loop), loop)
+                loop.run_until_complete(loop.shutdown_asyncgens())
+                loop.run_until_complete(
+                    loop.shutdown_default_executor(constants.THREAD_JOIN_TIMEOUT)
+                )
+            finally:
+                if self._set_event_loop:
+                    events.set_event_loop(None)
+                loop.close()
+                self._loop = None
+                self._state = _State.CLOSED
+
+        def run_app(
+            self,
+            app: Union[Application, Awaitable[Application]],
+            *,
+            host: Optional[Union[str, HostSequence]] = None,
+            port: Optional[int] = None,
+            path: Union[PathLike, TypingIterable[PathLike], None] = None,
+            sock: Optional[Union[socket.socket, TypingIterable[socket.socket]]] = None,
+            shutdown_timeout: float = 60.0,
+            keepalive_timeout: float = 75.0,
+            ssl_context: Optional[SSLContext] = None,
+            print: Optional[Callable[..., None]] = print,
+            backlog: int = 128,
+            access_log_class: Type[AbstractAccessLogger] = AccessLogger,
+            access_log_format: str = AccessLogger.LOG_FORMAT,
+            access_log: Optional[logging.Logger] = access_logger,
+            handle_signals: bool = True,
+            reuse_address: Optional[bool] = None,
+            reuse_port: Optional[bool] = None,
+            handler_cancellation: bool = False,
+        ) -> None:
+            """Run an app locally"""
+            self._lazy_init()
+
+            if (
+                self._loop.get_debug()
+                and access_log
+                and access_log.name == "aiohttp.access"
+            ):
+                if access_log.level == logging.NOTSET:
+                    access_log.setLevel(logging.DEBUG)
+                if not access_log.hasHandlers():
+                    access_log.addHandler(logging.StreamHandler())
+
+            main_task = self._loop.create_task(
```

Comment:
Should probably add the context parameter for compatibility:
https://github.com/python/cpython/blob/dc76a4ad3c46af3fb70361be10b8a791cd126931/Lib/asyncio/runners.py#L106-L109